python/tskit/trees.py

```diff
@@ -2004,9 +2016,17 @@ def get_leaves(self, u):
 
     def leaves(self, u=None):
         """
-        Returns an iterator over all the leaves in this tree that are
-        underneath the specified node. If u is not specified, return all leaves
-        in the tree.
+        Returns an iterator over all the leaves in this tree that are underneath
+        the specified node. If :math:`u`  is not specified, return all leaves on
+        the tree (i.e. all leaves reachable from the tree root(s), see note below).
+
+        .. note::
+            :math:`u` can be any node in the entire tree sequence, including ones
+            which are not connected via branches to a root node of the tree. If
+            called on such a node, the iterator will happily return "dead" leaves
+            (see :meth:`is_leaf`) which cannot be reached from a root of this tree.
```

Comment:
```suggestion
            called on such a node, the iterator will return the node itself, since it 
            is considered a "dead" leaf (see :meth:`is_leaf`) which cannot be reached from a root of this tree.
```

Reply:
That's not strictly true. You could also be calling `leaves()` on a node which is on a dead section of topology. In this case it will return all the dead leaves descending from that node on the (dead) topology.

Reply:
Ah, yes, you're right. OK, maybe just delete the "happily" then? This doesn't impart much information.

I  guess what you're doing in this case is manually specifying that "u" is a root for the purpose of this operation.